storage/google/cloud/storage/blob.py

```diff
@@ -631,13 +637,19 @@ def _do_multipart_upload(self, client, stream, content_type, size):
         upload_url = _MULTIPART_URL_TEMPLATE.format(
             bucket_path=self.bucket.path)
         upload = MultipartUpload(upload_url, headers=headers)
+
+        if num_retries is not None:
```

Comment:
Not in the private helper. The main entry point is `upload_from_file` and that does emit a warning